graphene_django/converter.py

```diff
@@ -313,14 +361,87 @@ def dynamic_type():
 @convert_django_field.register(models.OneToOneField)
 @convert_django_field.register(models.ForeignKey)
 def convert_field_to_djangomodel(field, registry=None):
+    from graphene.utils.str_converters import to_snake_case
+    from .types import DjangoObjectType
+
     model = field.related_model
 
     def dynamic_type():
         _type = registry.get_type_for_model(model)
         if not _type:
             return
 
-        return Field(
+        class CustomField(Field):
```

Comment:
As a general comment (and I don't mean this as a slight in any way), the code to achieve this behavior feels quite complex/confusing to me, bordering on it seeming like a code smell. e.g., are we properly handling all scenarios with it? How brittle is it? But I'm still not super familiar with all of the internals of graphene-django, so perhaps that's just how something of this lower-level nature would have to be written. I'd be curious to see @mahdyhamad's thoughts, since you're incorporating some of his efforts he'd described in https://github.com/graphql-python/graphene-django/pull/1315#discussion_r1125653751 but hadn't completed.

Reply:
I agree this needs to be tested further and maybe simplified/refactored. 
Also, it seems that this part from https://github.com/graphql-python/graphene-django/pull/1361 by-passes `get_queryset` when the resolver is awaitable. I plan to tackle this in a future PR. 

```python
if is_resolver_awaitable:
    fk_obj = resolver(root, info, **args)
    # In case the resolver is a custom awaitable resolver that overwrites
    # the default Django resolver
    return fk_obj
```